### PR TITLE
fix(nuxt): include method when creating `useFetch` auto key

### PR DESCRIPTION
**Pull request by maximepvrt**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by danielroe — APPROVED** [review]

**Review by TheAlexLichter — APPROVED** [review]
Even though the use case for anything else than `GET` is rather rare, it seems like a good idea 👍

**Comment by Hebilicious** [comment]
> Even though the use case for anything else than `GET` is rather rare, it seems like a good idea 👍

Fun fact, I recently learned that there's way more registered HTTP method than I realised https://www.iana.org/assignments/http-methods/http-methods.xhtml

File: packages/nuxt/src/app/composables/fetch.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ export function useFetch<
     return unref(r)
   })
 
-  const _key = opts.key || hash([autoKey, unref(opts.baseURL), typeof _request.value === 'string' ? _request.value : '', unref(opts.params || opts.query)])
+  const _key = opts.key || hash([autoKey, unref(opts.method), unref(opts.baseURL), typeof _request.value === 'string' ? _request.value : '', unref(opts.params || opts.query)])
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/fetch.ts:88]:
```suggestion
  const _key = opts.key || hash([autoKey, opts.method ? unref(opts.method).toUpperCase() : '', unref(opts.baseURL), typeof _request.value === 'string' ? _request.value : '', unref(opts.params || opts.query)])
```

Comment (manniL) [reply packages/nuxt/src/app/composables/fetch.ts:88]: Good catch! 👍

File: packages/nuxt/src/app/composables/fetch.ts [hunk 2]
```diff
@@ -85,7 +85,7 @@ export function useFetch<
     return unref(r)
   })
 
-  const _key = opts.key || hash([autoKey, unref(opts.baseURL), typeof _request.value === 'string' ? _request.value : '', unref(opts.params || opts.query)])
+  const _key = opts.key || hash([autoKey, opts.method ? unref(opts.method as MaybeRef<string>).toUpperCase() : 'GET', unref(opts.baseURL), typeof _request.value === 'string' ? _request.value : '', unref(opts.params || opts.query)])
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/fetch.ts:88]:
```suggestion
  const _key = opts.key || hash([autoKey, opts.method ? unref(opts.method as MaybeRef<string>)?.toUpperCase() || 'GET' : 'GET', unref(opts.baseURL), typeof _request.value === 'string' ? _request.value : '', unref(opts.params || opts.query)])
```

**Pull request merged**